### ISSUE TITLE
After respecing any character, it seems some scripts can fail

### ISSUE THREAD

**Issue by AgainPsychoX** [ThyWoof/SolastaRespec#4]
Spoilers of 4th/5th gem ahead, be aware.

There is fight with orc, that should go like this:
https://www.youtube.com/watch?v=gUCSK-vA1BA&start=185

You select character and it get teleported, unequipped and all other characters become NPCs. In my case, I couldn't proceed - either selected character was other than I selected or/and the selected character wasn't unequipped and other characters didn't became NPCs - which is crucial for the game script I guess.

I had Elizabeth and Edmund characters respec-ed, and I can't get to use them in the fight. The game script seems to work very chaotically, I tested 2 party configurations (orders):

```
Elizabeth > Marcus > Edmund > Fiora --> selecting Elizabeth --> Elizabeth get selected, cannot start fight; whole party is controllable 
Elizabeth > Marcus > Edmund > Fiora --> selecting Fiora --> Elizabeth get selected, cannot start fight; whole party is controllable 
Elizabeth > Marcus > Edmund > Fiora --> selecting Edmund --> Marcus get selected, deequiped (and can fight); only selected character is controllable
Elizabeth > Marcus > Edmund > Fiora --> selecting Marcus--> Fiora get selected, deequiped (and can fight); only selected character is controllable
Elizabeth > Fiora > Edmund  > Marcus --> selecting Elizabeth --> Elizabeth get selected, cannot start fight; whole party is controllable 
Elizabeth > Fiora > Edmund  > Marcus --> selecting Marcus --> Fiora get selected, deequiped (and can fight); only selected character is controllable
Elizabeth > Fiora > Edmund  > Marcus --> selecting Edmund --> Marcus get selected, deequiped (and can fight); only selected character is controllable
Elizabeth > Fiora > Edmund  > Marcus --> selecting Fiora --> Elizabeth get selected, cannot start fight; whole party is controllable
```

I post it as issue here, because I believe it's the respecing fault, as no one could help me on Solasta Discord, and I tried to use someone else save (with other characters, respec mod playthrough) and it worked "normally". I wonder, will game became completely stuck when someone decides to respec all 4 characters?

**Comment by ThyWoof** [comment]
which other mods do you have?

**Comment by ThyWoof** [comment]
I need a save game to test that.

**Comment by AgainPsychoX** [comment]
Mods I use:
![image](https://user-images.githubusercontent.com/23557839/124398403-d4d96200-dd15-11eb-9b23-f312afe296e6.png)

Last save before the fight I have is this:
[2 orcs.zip](https://github.com/ThyWoof/SolastaRespec/files/6760534/2.orcs.zip)

**Comment by ThyWoof** [comment]
This was probably fixed on 1.1.4 where I reuse the hero GUID. Unfortunately it won't fix an existing game
